static/usr/bin/fixrtc

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+NOW="$(date +'%s')"
+# XXX: also try /var/lib/systemd/timesync/clock ?
+MTIME_SELF="$(stat -L "/proc/self/exe" -c '%Y')"
```

Comment:
This is brilliant!!

Reply:
Actually, wait, I think you do not want `-L` here, otherwise you'll get the date of the script file.

Reply:
I think we need the `-L` - if this just stat()s /proc/self/exe it will read the 2000-01-01 (or 1970-01-01) date because the kernel time at this point is set to the rtc time which (at first boot) will be in the past. We need to move the time forward again and the baseline is the build/install time of the script on the filesystem. Or am I missing something?